tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -82,6 +110,17 @@ def parse_datafile_lines(datafile, program_type, section):
             continue
 
         schema = get_schema(line, section, schema_options)
+        if schema is None:
+            errors[line_number] = [util.generate_parser_error(
```

Comment:
They do but I felt that the `generate_error` function used there lost some key data that I have in my direct call. Looking over `make_generate_parser_error` used in those functions, do we lose information like record/field because they're being set to `None` in the call in `parse_datafile_line`? I'm open to removing this if clause but was just me solving a problem which based on the tests isn't a real problem? I'm not convinced it's needed after not touching it for awhile.

Perhaps I used this because I didn't "get" the nested-call situation with an intermediary nickname of `generate_error` that continues to catch me off.
`parse_datafile_line`-> `generate_error` -> `util.make_generate_parser_error` -> `generate_parser_error`

It's also a bit awkward to read:
```
if schema:
  ...
  return records
  
return [(None, False, [generate_error(...)])])
```


I also changed the error msg which I recall was feedback from Alex we had but that's a minor fix.

Reply:
i'm open to keeping the `field="Record_Type"` as you have it. you're correct though that the `record` is lost from the error - because it's a preparsing error, no record is created. no schema exists in this case either, which is why the field wasn't included initially. i believe the message was also updated as part of #2504. i'm fine with moving the check to this location to consolidate the two, but i'd recommend the existing ones be removed in that case.